### PR TITLE
Generate c++ code for complex numbers

### PR DESCRIPTION
**Pull request by namannimmo10**

### CODE REVIEW

**Comment by certik** [comment]
Does the generated C++ code compile?

File: tests/reference/cpp-complex2-cc31ff8.stdout [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        r(i) = v[i];
+    }
+    return r;
+}
+
+void test_complex_cpp()
+{
+    std::complex<float> c;
+    c = {423.000000, -23423.540000};
+    c = {1.000000, 2.000000};
```
Comment (namannimmo10) [inline tests/reference/cpp-complex2-cc31ff8.stdout:21]: This generated portion does compile, and I check that separately.

**Review by certik — APPROVED** [review]

File: tests/reference/cpp-complex2-cc31ff8.stdout (hunk 1 above)
Comment (certik) [reply tests/reference/cpp-complex2-cc31ff8.stdout:21]: Perfect!

**Comment by certik** [comment]
See also #77, which should make it much easier to test (we should allow generating code that only requires standard C++ compiler no 3rd party dependencies).

**Pull request merged**